zeus/optimizer/batch_size/client.py

```diff
@@ -93,20 +104,33 @@ def get_batch_size(self) -> int:
             params={"job_id": self.job.job_id},
         )
         self._handle_response(res)
+        parsed_response = TrialId.parse_obj(res.json())
 
-        batch_size = res.json()
-        if not isinstance(batch_size, int) or batch_size not in self.job.batch_sizes:
+        if parsed_response.batch_size not in self.job.batch_sizes:
             raise ZeusBSORuntimError(
-                f"Zeus server returned a strange batch_size: {batch_size}"
+                f"Zeus server returned a strange batch_size: {parsed_response.batch_size}"
+            )
+
+        self.current_batch_size = parsed_response.batch_size
+        self.trial_number = parsed_response.trial_number
+
+        logger.info(
+            "[BatchSizeOptimizer] Chosen batch size: %s", parsed_response.batch_size
+        )
+
+        def report_end() -> None:
+            # TODO: Double check capturing variable.
+            print(self.server_url + REPORT_END_URL, parsed_response.json())
+            httpx.patch(
```

Comment:
REST API doc said `POST` is when creating a new record, and `Patch` is for partial update of the resource (`Trial` in our case).

Reply:
I think the client is still POSTing independent units of information to those endpoints, and us inplace updating the Trial table is an internal implementation detail not exposed in the REST API. I would agree with patch of the client is sending in the entire row in the Trial table but with the modification of a couple fields, but that's not really the case.

Reply:
I think we should change a URL.
`/report` and make it a POST. Since we are creating a report resource. 
For ending,
`/trial` and make it a PATCH, since we are updating the status of the trial to finished.

Reply:
Sounds good!